test/controllers/AddClosureMetadataControllerSpec.scala

```diff
@@ -0,0 +1,182 @@
+      playStatus(addClosureMetadataPage) mustBe FOUND
+    }
+
+    "render an error if the api returns errors" in {
+      val consignmentId = UUID.fromString("c2efd3e6-6664-4582-8c28-dcf891f60e68")
+      val client = new GraphQLConfiguration(app.configuration).getClient[cm.Data, cm.Variables]()
+      val data: client.GraphqlData = client.GraphqlData(Option.empty, List(GraphQLClient.Error("Error", Nil, Nil, None)))
+      val dataString: String = data.asJson.printWith(Printer(dropNullValues = false, ""))
+      wiremockServer.stubFor(post(urlEqualTo("/graphql"))
+        .willReturn(okJson(dataString)))
+
+      val controller = instantiateAddClosureMetadataController()
+      val addClosureMetadataPage = controller.addClosureMetadata(consignmentId)
+        .apply(FakeRequest(GET, s"/standard/$consignmentId/add-closure-metadata").withCSRFToken)
+
+      val failure = addClosureMetadataPage.failed.futureValue
+      failure mustBe an[GraphQlException]
+    }
+  }
+
+  s"The consignment add closure metadata page" should {
+    s"return 403 if the GET is accessed by a non-standard user" in {
+      val consignmentId = UUID.fromString("c2efd3e6-6664-4582-8c28-dcf891f60e68")
+      val addClosureMetadataController = instantiateAddClosureMetadataController()
+
+      val addClosureMetadata = {
+        setConsignmentTypeResponse(wiremockServer, consignmentType = "judgment")
+        addClosureMetadataController.addClosureMetadata(consignmentId)
+          .apply(FakeRequest(GET, s"/consignment/$consignmentId/add-closure-metadata").withCSRFToken)
+      }
+      playStatus(addClosureMetadata) mustBe FORBIDDEN
+    }
+  }
+
+  private def instantiateAddClosureMetadataController(securityComponents: SecurityComponents = getAuthorisedSecurityComponents) = {
+    val graphQLConfiguration = new GraphQLConfiguration(app.configuration)
+    val consignmentService = new ConsignmentService(graphQLConfiguration)
+    val customMetadataService = new CustomMetadataService(graphQLConfiguration)
+
+
+    new AddClosureMetadataController(
+      securityComponents,
+      new GraphQLConfiguration(app.configuration),
+      getValidStandardUserKeycloakConfiguration,
+      consignmentService,
+      customMetadataService
+    )
+  }
+
+  private def mockGraphqlResponse() = {
+    val client: GraphQLClient[cm.Data, cm.Variables] = new GraphQLConfiguration(app.configuration).getClient[cm.Data, cm.Variables]()
+    val customMetadataResponse: cm.Data = getDataObject
+    val data: client.GraphqlData = client.GraphqlData(Some(customMetadataResponse))
+    val dataString: String = data.asJson.printWith(Printer(dropNullValues = false, ""))
+
+    wiremockServer.stubFor(post(urlEqualTo("/graphql"))
+      .withRequestBody(containing("customMetadata"))
+      .willReturn(okJson(dataString)))
+  }
+
+  private def getDataObject = {
+    // Until the 'sortMetadataIntoCorrectPageOrder', getDefaultValue and getFieldHints methods in the MetadataUtils are
```

Comment:
I've just noticed this comment so maybe what I said above about these methods might be redundant. I'm not sure why they're needed now though.